### PR TITLE
Enable Spotless on all Core libraries

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java [hunk 1]
```diff
@@ -40,37 +40,26 @@ public class ProxyOptions implements AutoCloseable {
...
-
-    private static final ConfigurationProperty<String> USER_PROPERTY = ConfigurationPropertyBuilder.ofString(ConfigurationProperties.AMQP_PROXY_USER)
-        .shared(true)
-        .logValue(true)
-        .build();
-
-    private static final ConfigurationProperty<String> PASSWORD_PROPERTY = ConfigurationPropertyBuilder.ofString(ConfigurationProperties.AMQP_PROXY_PASSWORD)
-        .shared(true)
-        .build();
+    private static final ConfigurationProperty<ProxyAuthenticationType> AUTH_TYPE_PROPERTY
```
Comment (joshfree) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java:43]: It's a bit weird that ConfigurationPropertyBuilder isn't always on the same line

**Comment by joshfree** [comment]
spot checked for regressions

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpMessageConstant.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
  * Reserved well-known constants from AMQP protocol.
  *
  * @see <a href="https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-properties">
- * AMQP 1.0: Messaging Properties</a>
+ *      AMQP 1.0: Messaging Properties</a>
```
Comment (srnagar) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpMessageConstant.java:14]: nit: this spacing can be removed.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpRetryOptions.java [hunk 3]
```diff
@@ -194,8 +194,7 @@ public boolean equals(Object obj) {
 
         final AmqpRetryOptions other = (AmqpRetryOptions) obj;
 
-        return this.getMaxRetries() == other.getMaxRetries()
-            && this.getMode() == other.getMode()
+        return this.getMaxRetries() == other.getMaxRetries() && this.getMode() == other.getMode()
```
Comment (srnagar) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpRetryOptions.java:197]: nit: I liked the previous formatting better as it kept all the conditions in new lines. Now, couple of them are in the same line and the rest are in new lines.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpRetryPolicy.java [hunk 4]
```diff
@@ -69,17 +69,15 @@ public int getMaxRetries() {
...
             || retryCount > retryOptions.getMaxRetries()) {
             return null;
         }
 
         final Duration baseDelay;
         if (lastException instanceof AmqpException && isRetriableException(lastException)) {
             baseDelay = ((AmqpException) lastException).getErrorCondition() == SERVER_BUSY_ERROR
-                ? retryOptions.getDelay().plus(SERVER_BUSY_WAIT_TIME)
-                : retryOptions.getDelay();
+                ? retryOptions.getDelay().plus(SERVER_BUSY_WAIT_TIME) : retryOptions.getDelay();
```
Comment (srnagar) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpRetryPolicy.java:80]: For tertiary conditions, I think we should make the `if` and `else` case be in separate lines to be consistent and easy to ready.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpChannelProcessor.java [hunk 5]
```diff
@@ -450,8 +434,7 @@ public void onError(Throwable throwable) {
             if (!isCancelled()) {
                 processor.subscribers.remove(this);
                 actual.onError(throwable);
-                processor.logger.atInfo()
-                    .addKeyValue(SUBSCRIBER_ID_KEY, subscriberId)
+                processor.logger.atInfo().addKeyValue(SUBSCRIBER_ID_KEY, subscriberId)
```
Comment (srnagar) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpChannelProcessor.java:437]: Should we make all chained calls (>=3) be on separate lines?

**Review by srnagar — APPROVED** [review]
I had some minor comments about formatting but it's fine if those cannot be configured in spotless.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpChannelProcessor.java (hunk 5 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/AmqpChannelProcessor.java:437]: Unfortunately, this only has a few configuration options


Always newline

```java
process
    .logger
    .atInfo()
    .addKeyValue(SUBSCRIBER_ID_KEY, subscriberId)
```


Split after first

```java
process.logger
    .atInfo()
    .addKeyValue(SUBSCRIBER_ID_KEY, subscriberId)
```


Current behavior / split only required

```java
 processor.logger.atInfo().addKeyValue(SUBSCRIBER_ID_KEY, subscriberId)
```

From looking at this before the first two make the code very long, which necessarily isn't a problem. But it's almost the inverse terseness issue where it's too spread out.

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpMessageConstant.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpMessageConstant.java:14]: Hmm, never seen this before, will look into what setting causes this

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpRetryOptions.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/AmqpRetryOptions.java:197]: I believe this and the conditionals should be doable

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/ProxyOptions.java:43]: This is due to the constructor length, Eclipse formatting split based on the first call. The cases where the constructor call is longer than the remaining line space forces it to another line as it's preferring to keep all constructor arguments on as few lines as possible.

The cases where it's on the same line there isn't a first call too long to fit on the line

**Pull request merged**